### DISCUSSION TITLE
Best practices for running callable script

### DISCUSSION THREAD

**Discussion by joneslloyd** [vercel/next.js#12548] [Help] [answered]
Hello,

I was just wondering what the recommended way to run a 'callable' script is, in my Next.js app?

To be clear, this isn't really an API endpoint that gets/sets data, rather a Node.js script that I will run daily (via AWS Lambda or some a cron service) and add some records to the database. But it's also definitely not a 'page' that I want to be publicly searchable.

I've so far just added the script to the `/pages/api/` folder, and it all works perfectly (I just export a function without `req` or `res` parameters), but I wanted to check whether this is the only/recommended way of doing things.

Thanks in advance!

**Comment by brianlovin** [comment, marked as answer]
It could be worth investigating migrating something like this to a dedicated GitHub action or external service. If you have this function as a page in `pages/api/` it's accessible to the internet, so you'd need some kind of authentication/authorization rules, or risk it getting spammed and doing a lot of extra work on your DB.

**Reply by joneslloyd** [reply]
Great suggestion!

My thinking was that, because this is a relatively small process, I wouldn't need to go the bother of moving it offsite (but would make sure it's authenticated).

If I wanted to keep this script as part of the site, is my current method probably the best way of doing it?

Thanks!

**Reply by brianlovin** [reply]
If you're fine with it being exposed to the internet, then sure! I'm not confident it's the "best way" - but hey, if it works that's a good start. This still feels like something more suited to a separate lambda or even a GitHub Action, but I understand that's a bit more effort :)

**Reply by joneslloyd** [reply]
You know what? I think you've persuaded me to do this the correct way, even if the particular task is a small one! 🙂

**Comment by tubbo** [comment]
I also want to do this but for developer-centric scripts like seeding the database. I was trying to use my Mongoose models using a simple `node tasks/seed.js` script but that didn't seem to work due to the fact that my models are written in Babel ES6 and not the ES6 module syntax that Node expects. Trying to change them to match Node's expectations also proved fruitless. 

This is what I get when I try running the file using `babel-node` and the **next/babel** preset:

```
$ babel-node --presets=next/babel tasks/seed.js
~/Code/nextapollo/node_modules/babel-core/lib/transformation/file/options/option-manager.js:328
        throw e;
        ^

TypeError: api.caller is not a function (While processing preset: "~/Code/nextapollo/node_modules/next/babel.js")
```

Not really sure how to proceed...
